src/main/java/com/social/accesstoken/services/PermissionService.java

```diff
@@ -0,0 +1,77 @@
+package com.social.accesstoken.services;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.NodeList;
+
+import com.gargoylesoftware.htmlunit.html.HtmlButton;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlSubmitInput;
+import com.social.utils.AutorizeDictionary;
+
+/**
+ * Service help you automatically login on the website and get rule
+ * @author salazart
+ *
+ */
+public class PermissionService {
```

Comment:
I think you need to refactor this class, as  it is not so easy to understand. In my oppinion you don't need getPermissionForm and emulatePermissionButtonClick methods, you can have one method instead of them, and other methods are not so clean. Try to rewrite them in some other simple way with javadoc